ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -191,10 +28,36 @@ def add_arguments(self, parser):
             "The default of -1 considers ALL documents.",
         )
 
+    def iter_solr(self, nsize=10, item_type='page'):
+        i=0
+        q=SolrQuerySet().search(item_type=item_type)
+        total = q.count()
+        for i in progressbar(range(0, total, nsize)):
+            q.set_limits(i,i+nsize)
```

Comment:
why do you need to set limits every iteration? should this be an update call? (could be I've forgotten how to use progressbar, it's been a while)

Reply:
while I'm thinking about it: would be good to make progressbar optional, since it's possibly we'll want to run this script without that (e.g. if we set it up as a cron job)

Reply:
the `set_limits` is from parasolor as method of the SolrQuerySet. progressbar ended up being simple and nearly identical to tqdm. will add an verbose/quiet flag

Reply:
ah, I forgot about that syntax because I never use it directly. SolrQuerySet is designed to support indexing like a list, so you should be able to do `[100:200]` to get a chunk of results and it should query Solr for you.  

quite possible this is not documented or not clearly documented (please flag/help resolve)